src/cargo/core/summary.rs

```diff
@@ -25,6 +25,7 @@ struct Inner {
     features: Rc<FeatureMap>,
     checksum: Option<String>,
     links: Option<InternedString>,
+    rust_version: Option<InternedString>,
```

Comment:
From another angle, should we put `rust-version` outside JSON blob? This is how [`SummariesCache::parse`](https://github.com/rust-lang/cargo/blob/0ad289b264f6003a663832103e2ea810e8556dbf/src/cargo/sources/registry/index.rs#L705) does for crate `version` to avoid unnecessary JSON blob parsing. This should have benefits given that we want a MSRV-aware resolver, though undeniably it makes things way more complicated.

Reply:
Unless I misunderstood, I believe by the point we want the `rust-version`, we'll have loaded the blob.  The plan isn't to pre-filter the rust-version but to add the `rust` as a new kind of dependency.

Reply:
I see. Thank you!

Just re-read the thread. Treat `rustc` as a kind of implicit dependency. Brilliant!